### PR TITLE
Remove gcc legacy scripts

### PR DESCRIPTION
**Pull request by solotzg**
### What problem does this PR solve?

Issue Number: ref #6233, ref #6201

Problem Summary:

Since we do not support gcc any more, this PR will clean up gcc legacy scripts.

### What is changed and how it works?

- set (CLANG_MINIMUM_VERSION 12)
- remove gcc condition in cmake scripts.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

**Comment by SeaRise** [comment]
Do we need to update readme.md?

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,36 +59,28 @@ if (NOT MSVC)
...
+    set (CLANG_TIME_TRACES_FLAGS "-ftime-trace")
+    message (STATUS "Using clang time traces flag `${CLANG_TIME_TRACES_FLAGS}`. Generates JSON file based on output filename. Results can be analyzed with chrome://tracing or https://www.speedscope.app for flamegraph visualization.")
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CLANG_TIME_TRACES_FLAGS}")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CLANG_TIME_TRACES_FLAGS}")
 endif ()
 
-if (COMPILER_CLANG)
-    # Clang doesn't have int128 predefined macros, workaround by manually defining them
-    # Reference: https://stackoverflow.com/questions/41198673/uint128-t-not-working-with-clang-and-libstdc
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__GLIBCXX_BITSIZE_INT_N_0=128 -D__GLIBCXX_TYPE_INT_N_0=__int128")
```
Comment (JaySon-Huang) [inline CMakeLists.txt:69]: Seems these lines are removed?

Comment (solotzg) [reply CMakeLists.txt:69]: There is no such compilation problem since we use clang >= 12 with flag `-std=gnu++17`.

Comment (JaySon-Huang) [reply CMakeLists.txt:69]: OK

**Review by JaySon-Huang — APPROVED** [review]

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**